src/fluree/db/commit/storage.cljc

```diff
@@ -38,13 +38,29 @@
 (defn read-verified-commit
   [storage commit-address]
   (go-try
-    (when-let [commit-data (<? (storage/read-json storage commit-address))]
-      (log/trace "read-commit at:" commit-address "data:" commit-data)
-      (let [addr-key-path (if (contains? commit-data "credentialSubject")
-                            ["credentialSubject" "address"]
-                            ["address"])]
-        (-> commit-data
-            (assoc-in addr-key-path commit-address)
+    (when-let [commit-json (<? (storage/read-json storage commit-address))]
+      (log/trace "read-commit at:" commit-address "data:" commit-json)
+      (let [cred?         (contains? commit-json "credentialSubject")
+            subject-path  (if cred? ["credentialSubject"] [])
+            addr-path     (conj subject-path "address")
+            subject-json  (if cred? (get commit-json "credentialSubject") commit-json)
+            ;; Prefer commit id from address/hash; fallback to computing from JSON only if needed
+            existing-id   (get subject-json "id")
+            ;; Extract base32 hash-like token from the address if present
+            addr-str      (str commit-address)
+            hash-token    (last (clojure.string/split addr-str #"/"))
+            normalized-b  (when (and hash-token (clojure.string/starts-with? hash-token "b")) hash-token)
+            id-from-addr  (when normalized-b (str "fluree:commit:sha256:" normalized-b))
+            computed-id   (or id-from-addr (commit-data/commit-json->commit-id subject-json))
```

Comment:
There should be a separate function to compute the id if necessary. This let form is hard to follow because there are so many bindings doing so many things in one place.

Reply:
There is in fact a recently added [`hash->commit-id` function](https://github.com/fluree/db/blob/0ce80297e61f90cb59dffd4538b79e302fb48e60/src/fluree/db/flake/commit_data.cljc#L197-L199)